### PR TITLE
Refactored dt tap test to remove async lib

### PR DESCRIPTION
**Pull request by mrickard**
Signed-off-by: mrickard <maurice@mauricerickard.com>

## Proposed Release Notes
Removed async library from distributed tracing integration test

## Links
https://issues.newrelic.com/browse/NR-35280

### CODE REVIEW

File: test/integration/distributed-tracing/dt.tap.js [hunk 1]
```diff
@@ -250,71 +266,33 @@ tap.test('distributed tracing', (t) => {
...
+      const tx = agent.tracer.getTransaction()
+      tx.nameState.appendPath('foobar')
+
+      return get(generateUrl(port, endpoint), (err, body) => {
+        tx.nameState.popPath('foobar')
+        createResponse(req, res, body, bodyProperty)
+      })
+    }
+
+    start = await generateServer(http, api, START_PORT, cb, (req, res) => {
```
[the remark this answers is not in the document]
Comment (mrickard) [reply test/integration/distributed-tracing/dt.tap.js:279]: Looks like they actually aren't!

[the remark this answers is not in the document]
Comment (mrickard) [reply test/integration/distributed-tracing/dt.tap.js:279]: It does look to me, though, that generateServer does return a server, which is how we can invoke those close methods later. This test structure...does rely a lot on side effects, so some return values matter, but a lot don't.

File: test/integration/distributed-tracing/dt.tap.js [hunk 2]
```diff
@@ -237,6 +245,14 @@ tap.test('distributed tracing', (t) => {
 
   t.autoend()
 
+  // simulation of the callback used by the async library, used by generateServer and close
+  const cb = (err, result) => {
```
[the remark this answers is not in the document]
Comment (mrickard) [reply test/integration/distributed-tracing/dt.tap.js]: Fixed

File: test/integration/distributed-tracing/dt.tap.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (mrickard) [reply test/integration/distributed-tracing/dt.tap.js]: `await`s removed.

File: test/integration/distributed-tracing/dt.tap.js [hunk 3]
```diff
@@ -250,71 +266,33 @@ tap.test('distributed tracing', (t) => {
...
   t.afterEach(async () => {
     helper.unloadAgent(agent)
 
-    await new Promise((resolve) => {
-      async.parallel(
-        [(cb) => start.close(cb), (cb) => middle.close(cb), (cb) => end.close(cb)],
-        resolve
-      )
-    })
+    await start.close(cb)
```
[the remark this answers is not in the document]
Comment (mrickard) [reply test/integration/distributed-tracing/dt.tap.js]: It's nice to have this as a one-liner. Done.

**Review by bizob2828 — APPROVED** [review]

**Pull request merged**